### ISSUE TITLE
Disk activity indicators for drives in Slot 5

### ISSUE THREAD

**Issue by skiphipdata** [AppleWin/AppleWin#1134]
It would be nice to have a separate set of disk activity "LEDs" for the drives in Slot 5.

**Comment by tomcw** [comment]
This feels like a useful think to add, so let me share some work in progress...


I added extra "Slot 5" Disk-II LEDs & track/sector status, below the "Slot 6" Disk-II status:
(Only visible in 2x Windowed mode)
![image](https://user-images.githubusercontent.com/6696896/196029306-ec2787d1-f5b0-42e0-9f41-b044be6fd8c1.png)

But I felt there was too much track/sector debug "noise" for a regular user, so I made it only visible via this extra config checkbox:
NB. the "Slot 6" track/sector status is always visible in 2x Windowed mode)
![Config-Disk](https://user-images.githubusercontent.com/6696896/196029327-53cf3df2-e3d2-46f7-ab05-52328ed9bda2.png)

**Comment by burniouf** [comment]
wouldn't it be possible to display also exact Quarter track position for all Disk ][ drives ?

**Comment by tomcw** [comment]
@burniouf - using this small font, this is what the characters: ½, ¼ and ¾ look like.
IE. the image below is meant to display the following:

T 3½  T 0¼
S 6¾ S ??

![image](https://user-images.githubusercontent.com/6696896/197408058-9636beec-e050-4ed8-9116-21576f3cd3e3.png)

Given the font size, I don't think those fractions are legible.

But since they are changing so quickly - do you care?

Other ideas (just focussing on the current UI layout for now)
- have a mouse-over tooltip to show the full decimal track, eg. "T 3.75"
- use a simple replacement character, eg. "a","b","c" to mean ".25",".5",".75", eg. "T 3c" to mean "T 3.75"
- use a better glyph, eg. -,=,三, eg. "T 3三" to mean "T 3.75"
  - or 1/2/3 dots, eg, . : ⋮  , eg. "T 3⋮" to mean "T 3.75"

**Comment by sicklittlemonkey** [comment]
Nice work Tom - I like the ability to simplify the display.
I think .25 etc is clearest, but in case you wanted other ideas, you could also use [symbols](https://www.w3.org/TR/xml-entity-names/025.html).

Quarters of a block: ▘ , ▚ , and ▛ or ▜ .
There are also quarters of a circle: ◔ , ◑ , ◕ etc.

**Comment by tomcw** [comment]
Thanks.

btw, you can already see quarter track (or half phase) info from the built-in debugger (F7), using `disk i[nfo]`:

![image](https://user-images.githubusercontent.com/6696896/197832380-8d4b5f3e-8b12-4558-8947-9b0caaa0c84a.png)

(but currently only for slot-6's enabled drive; and only quarter track precision for .woz images)

**Comment by burniouf** [comment]
just read again above.....
your idea of "have a mouse-over tooltip to show the full decimal track, eg. "T 3.75" could be fine !
you may add in the tooltip T$00,25, ph0:on/off, ph1:on/off,ph2:on/off,ph3:on/off....bitOffset....etc...
"phase $42.50" has not a clear meaning, to my opinion

**Comment by tomcw** [comment]
Thanks @burniouf - I agree, a tooltip gives more scope for richer details and means I don't need to try to "shoehorn" more into the existing UI's tiny space.

**Comment by tomcw** [comment]
I'm experimenting with:
- Slot 5 Disk II LEDs are always visible (if Slot 5 card is diskii)
- A UI toggle: used to switch between showing the track/sector info for **both** slots _or_ no track/sector info (just show LEDs)
- The tooltip (which is WIP) currently shows hex & dec track info for both drives

Left is: show full track/sector info for all s6+s5 drives; and right is: just show LEDs:
![image](https://user-images.githubusercontent.com/6696896/199840619-08ea0155-1a0d-4500-ac1b-34707e396f59.png) ![image](https://user-images.githubusercontent.com/6696896/199842162-a447692f-9eee-47bf-b8a6-95668d033c56.png)

**Comment by sicklittlemonkey** [comment]
Looks good! My only requested change would be to add spaces, i.e. "Drive1" -> "Drive 1". ; - )

**Comment by burniouf** [comment]
Same for me, looks perfect! if you want to debug easily with Qtracks, just boot ProDOS.FX media and use a 192k formatted disk

**Comment by tomcw** [comment]
Re-orientate, so that drive 1 (and drive 2) info is vertical - so it matches the GUI that the tooltip is for:

![image](https://user-images.githubusercontent.com/6696896/200080466-39f75c32-03ed-4bd1-8079-e0ff2ede8292.png)

**Comment by tomcw** [comment]
Again drive 1 (and drive 2) info is vertical - but this time with track's hex and dec on a single (longer line), and then sector's hex/dec on next line:

![image](https://user-images.githubusercontent.com/6696896/200081755-cdccfdae-132c-4d0b-91c7-5fa4d4c8f771.png)

**Comment by tomcw** [comment]
Probably this latter one is best?
btw, if we know it's a DOS (or ProDOS) disk, then we could also include the logical sector number too (or block number for ProDOS).

**Comment by Michaelangel007** [comment]
> Re-orientate, so that drive 1 (and drive 2) info is vertical - so it matches the GUI that the tooltip is for:
> 
> ![image](https://user-images.githubusercontent.com/6696896/200080466-39f75c32-03ed-4bd1-8079-e0ff2ede8292.png)

I like that both Hex and Dec values are shown!

Another option might be to drop the the `T` and `S` entirely for the hex display and just put the hex values on one line?

`$11.00/F`

**Comment by sicklittlemonkey** [comment]
Slightly off topic: I don't think I ever saw hex used for the rational quarter track part, ie: $11.4 ... $11.8 ... $11.C.
It's a pity, as it's nice and compact.

**Comment by tomcw** [comment]
Thanks for the further suggestions (and yes, a shame the hex fractional part didn't ever catch on).

In the end I've settled on this wide tooltip:

![image](https://user-images.githubusercontent.com/6696896/200081755-cdccfdae-132c-4d0b-91c7-5fa4d4c8f771.png)

**Comment by tomcw** [comment]
btw, I also extended the debugger to support `DISK` commands for Disk II cards in other slots:

```
Debugger: add 'DISK SLOT n' - to set current slot of Disk II card
  - ie. support 'DISK INFO' for slot 5
```

**Comment by medasaro** [comment]
I, too, would very much like to have a display for fractional tracks. It is very useful when debugging why copy-protected .woz files won't work.

-Matthew D'Asaro

**Comment by tomcw** [comment]
Released this in [AppleWin 1.30.13.0](https://github.com/AppleWin/AppleWin/releases/tag/v1.30.13.0), so closing this issue.

**Comment by tingtron** [comment]
On some ProDOS disks
I am getting `??` in `S` when track `T` is getting updated.
Compared with a previous version of AppleWin this does not happen -- it always shows `S` as non-`??` whenever `T` is updated.
